authors/apps/article_rating/views.py

```diff
@@ -0,0 +1,62 @@
+from django.db.models import Avg
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import status
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from authors.apps.articles.models import Article
+from authors.apps.article_rating.models import ArticleRating
+from authors.apps.article_rating.serializers import ArticleRatingSerializer
+
+class AverageArticleRatingViewSet(ModelViewSet):
+    """Endpoint for getting the average rating of a given article"""
+    permission_classes = (IsAuthenticated,)
+    queryset = ArticleRating.objects.all()
+    serializer_class = ArticleRatingSerializer
+
+    def list(self, request, slug):
+        article = Article.objects.get(slug=slug)
+        queryset = ArticleRating.objects.all().filter(article=article)
+        serializer = ArticleRatingSerializer(queryset, many=True,context={'request': request})
+        average = queryset.aggregate(Avg('rating'))
+        return Response({"rating": average.get("rating__avg")})
```

Comment:
Is it okay if you add more info, for example, the article title in the response for average article rating
<img width="1440" alt="Screenshot 2019-04-29 at 12 57 39" src="https://user-images.githubusercontent.com/24764476/56889404-f0ce8280-6a7e-11e9-832c-2ebabc4d6a8d.png">
```
{
   "title": "article title",
   "average_rating": 3
}
```
Also modify the key from rating to average rating.
Good progress @llwasampijja

Reply:
That would be ok @kalsmic !